runtime/v1/shim/client/client.go

```diff
@@ -188,15 +188,30 @@ func annonDialer(address string, timeout time.Duration) (net.Conn, error) {
 }
 
 // WithConnect connects to an existing shim
-func WithConnect(address string, onClose func()) Opt {
+// For compatible purpose we use a list of addresses, but only one of them should be valid
+func WithConnect(addresses []string, onClose func()) Opt {
 	return func(ctx context.Context, config shim.Config) (shimapi.ShimService, io.Closer, error) {
-		conn, err := connect(address, annonDialer)
-		if err != nil {
-			return nil, nil, err
+		var (
+			conn net.Conn
+			err  error
+		)
+		for _, address := range addresses {
+			conn, err = connect(address, annonDialer)
+			if err != nil {
+				nerr, ok := err.(net.Error)
+				// It implies the address is valid,
+				// so we didn't continue
+				if ok && nerr.Timeout() || nerr.Temporary() {
```

Comment:
if the `ok` is false, it will panic. could you mind to make the condition more clearly?